Automattic/jetpack
# Jetpack: handle VPBlock notices via withNotices HOC

This PR re-implements the error handling in the UI using the withNotices() HOC and also ensures to show the error produced by the Mediaplaceholer instance that's created by the uploader component.

Fixes #

#### Changes proposed in this Pull Request:

* Jetpack: handle VPBlock notices via withNotices HOC

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Go to the block editor
* create a VPBlock instance
* try to generate different errors. Confirm the error notices shows properly:

Error | Screenshot
------|------
Unsupported file type | <img width="666" alt="image" src="https://user-images.githubusercontent.com/77539/178518001-3b05c60a-4698-4f52-bc42-42cb8061d868.png">
Trying to upload from an invalid host | <img width="663" alt="image" src="https://user-images.githubusercontent.com/77539/178518145-ddf5d569-b60e-48eb-aab4-0001e40ce13f.png">
Trying to link a private VideoPress video | <img width="663" alt="image" src="https://user-images.githubusercontent.com/77539/178518490-a7222512-fba0-4904-89de-19ab44926cab.png">






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202586735461847
